src/main/java/com/cloudbees/plugins/deployer/sources/MavenArtifactDeploySource.java

```diff
@@ -158,7 +158,7 @@ public String getGroupId() {
      */
     @NonNull
     public String getType() {
-        return type == null ? DEFAULT_TYPE : type;
+        return type;
```

Comment:
Object deserialization does not involve the constructor. So, if somebody migrates old data, this field can be probably null